### ISSUE TITLE
WS252 contained a new strain of C. remanei px356 - web dev for access points to the data.

### ISSUE THREAD

**Issue by Paul-Davis** [WormBase/website#4459]
Hi,

One thing included in the WS252 release is the incorporation of a new strain of C. remanei (px356) PRJNA248909

I can see the data on the wormbase ftp, but it looks like we need to build in the access points on the WS252 website:
Species drop down - Probably best placed in the Species -> Other -> C. remanei (px356) position rather than having it under the core C. remanei.
GBrowse
JBrowse
etc.

Thanks

Paul

**Comment by scottcain** [comment]
GBrowse for this assembly is built: http://staging.wormbase.org/tools/genome/gbrowse/c_remanei_PRJNA248909/

Please take a look and let me know if it can be improved (better examples or initial landmarks for instance).

JBrowse would be done now as well, but I'm simultaneously working on a new release of the JBrowse software which is presenting some challenges.  Because that didn't go as smoothly as I'd hoped, I'll build the entire JBrowse instance with the old version of the software to make that available on staging while I work on the new version.  

tl;dr: JBrowse will be available for the new assembly later today.

**Comment by scottcain** [comment]
The JBrowse instance for this assembly is also now available on staging:

http://staging.wormbase.org/tools/genome/jbrowse/full.html?data=data%2Fc_remanei_PRJNA248909

**Comment by Paul-Davis** [comment]
Hi @scottcain, Thanks for getting the browsers up, is there any means of getting feature labels for this level of genome? even if it was just the "FL81_00021" style names displayed above/below the names. This could be made standars for all genomes of thie "Tier" is it is possible.

Thanks

**Comment by sibyl229** [comment]
Hi @Paul-Davis I'm working providing a website endpoint for C. remanei (px356) and objects associated with it. I wonder if you could help me with a problem I ran into. The problem I have is to identify an object as one for C. remanei (px356) based on data in ACe dumps. This is necessary to allow Browse and Basic Search on the species index page to work correctly. Otherwise the C. remanei (px356) genes get mixed up with other C. remanei genes in search result. 

Currently search index is built from ACe dumps, but the information there doesn't explicit specify the Bioproject ID. It seems that the WB ID of the C. remanei (px356) genes are prefixed with "PRJNA248909:", I wonder if this is a reliable way to identify the bioproject, and whether it will likely change in the future.

Thank you very much!

**Comment by Paul-Davis** [comment]
Hi @sibyl229,

The Bioproject should not change for this strain. Which ace dumps are you refering to, is this something that gets done at OICR?

Genes have the "FL81" prefix as a standard identifier.

Sorry I can't be of more help.

**Comment by epaule** [comment]
on that note: you might have the same problem with the C.elegans Hawaii strain

**Comment by sibyl229** [comment]
@Paul-Davis Thanks. Hmm do you know if the Bioproject ID is in ACeDB at all for objects like genes? And if not, do you know if it makes sense to include it sometime in the future?

@epaule you are right! Thanks for the reminder. Is there a way to identify those Hawaiian strain genes based on information in ACeDB?

**Comment by sibyl229** [comment]
Hi @scottcain It seems that the gbrowse URL for C. remanei genes (both strains) gives me 'Not found' :( Could you take a look when you have time? 

Here are the links I tried:
http://staging.wormbase.org/tools/genome/gbrowse/c_remanei_PRJNA248909?name=gene:PRJNA248909:FL81_00001
http://staging.wormbase.org/tools/genome/gbrowse/c_remanei_PRJNA53967?name=gene:WBGene00000790

**Comment by scottcain** [comment]
Hi @sibyl229 those urls seem a little sketchy.  The second one gives me a not found, but then when I search WB for it, I find that the gene is dead and has no sequence features, so I guess I'm not surprised that it can't be found in GBrowse.  For the first one, if I change the name argument to "FL81_00001" instead of "gene:PRJNA248909:FL81_00001" it does get found.  Should that be how the URL is written, or is that gene:bioproject:name construct a common thing for the other species?

**Comment by epaule** [comment]
sorry, @sibyl229 my bad, the Hawaii strain only has projected genes, which do not appear in the ACeDB database.

**Comment by Paul-Davis** [comment]
@sibyl229 The new C. remanei Strain is set up a bit like thr original remanei reference strain so all the info is in the database.

?Species Caenorhabditis remanei
Has conected to it 2 x ?Sequence_collections

Genome:C_remanei-15.0.1 - Reference strain
Genome:C_remanei_Oregon_v1 - New Strain

In here you have the NCBI_BioProject PRJNA248909 and other generally useful information about who produced it etc.

**Comment by sibyl229** [comment]
Hi @Paul-Davis Thanks for the additional information. Here's how I understand the model, and it looks like a gene object isn't identified with a bioproject in ACeDB because it's not uniquely associated with a Sequence_collection:

Gene object -> Species ( one-to-one)
Species -> Sequence_collection **(one-to-many)**
Sequence_collection -> bioprojcet (one-to-one)

Does it makes sense to have a gene unambiguously associated with a Sequence_collection?

No problem @epaule. Thanks for the clarification!

**Comment by epaule** [comment]
there is a Strain tag in the Gene class that we could populate

**Comment by sibyl229** [comment]
@epaule The same problem goes beyond Gene class, and potentially extends to every class that's currently associated with a species. You guys probably need time to decide if that's the direction to go, and I don't think there's a rush implementing anything yet. For now, I hacked around the c_remanei_PRJNA248909 vs c_remanei reference problem, so it work correctly with filtered search and browse by strain (bioproject). But I imagine it will get messier without a standard way to determine the strain or bioproject an object comes from, when more strains and their associated objects are added.

**Comment by Paul-Davis** [comment]
Hmmm an interesting point, we could make features more explicitly connected with the assembly that they belong to?!

**Comment by sibyl229** [comment]
@Paul-Davis That will work for my use case :)

**Comment by sibyl229** [comment]
Hi @Paul-Davis The C. remanei px356 strain is added to the species dropdown with links to gbrowse. Here is the species index page for  C. remanei px356:  http://staging.wormbase.org/species/c_remanei_PRJNA248909#301-2-10 

Previously there weren't access points (or index pages) for additional assemblies of a species, so this this this page probably requires more thorough testing. I tested it, but it would be great if you could double check the links in Download, Genome Assemblies, Search and Browse widgets. 

Additionally the C. remanei index page should show the default assembly PRJNA53967: http://staging.wormbase.org/species/c_remanei#301-2-10 
Thank you!

**Comment by khowe** [comment]
@sibyl229 I see your point. The intended model is that ?Sequence_collection refers to the set of ?Sequence objects in the collection, as well as a set of properties for the collection (including BioProject). 

```
?Sequence_collection Origin Evidence #Evidence
.
.
.
                                    Sequences ?Sequence
```

A sequence feature (?Gene, ?Transcript etc), will be assocated with a parent sequence, which will belong to exactly one ?Sequence_collection. So to find the BioProject for a Gene, the logic would be:

(a) Get parent ?Sequence _s_ for gene;
(b) Identify ?Sequence_collection _sc_ that _s_ belongs to 
(c) Get Bioproject from _sc_

However, a long time ago we decided not to create ?Sequence objects in Acedb tier3 assemblies, because we thought that nothing would need to use them and it would save space. So, to allow you to look up the information that you need, we could add in these ?Sequence objects. We could create them as "stubs" without actual sequence, to save space. Note also that to make (b) above efficient, we would probably need a models tweak (to add a two-way XREF between ?Sequence and ?Sequence_collection - which is logically the correct thing to do anyway I think, regardless of this issue). 

However, your last post suggests that you have found a workaround. Is it a clean one? Or should we go ahead with the fixing the data/model?

**Comment by sibyl229** [comment]
Hi @khowe thanks for the detailed explanation! The solution you described with Sequence stub and two-way link between ?Sequence and ?Sequence_collection will work for me (with search indexing and display). A minor question: in short term, we don't need to worry about non- sequence-feature type classes that's associated to a species (like  Expression_cluster, Microarray_results), right?

The workaround I used relies on the WormBase ID of tier3 objects having the BioProject ID as prefix. If all the tier 3 objects' WBID follow this convention, it's also sufficient for me to identify their BioProject and index them with the correct species.

So either way works for me, you guys can pick whichever way makes more sense for you.

**Comment by khowe** [comment]
Hi @sibyl229 . All of the "tier3" genes and proteins will have bioproject: prefixes, so let's stick with that as the solution for now, rather than filling up the database with more objects and connections. 

For other classes, it's a bit more fiddly to determine which genome project (bioproject) the object relates to. For some, like ?Expression_cluster, the path back to a parent sequence is complicated. 

However, for population of the menus, all genomes outside the "core" set (at the top level in the menu) should have only have three things in the menu: 

Genome Browser
Genes
Proteins

**Comment by sibyl229** [comment]
Hi @Paul-Davis are you okay with the access points for C. remanei px356 through the drop down menu? Shall I close this issue? Thanks!

**Comment by Paul-Davis** [comment]
Hi @sibyl229 yes I guess it is consistent with the C. elegans Hawaii strain so that's fine to close.
